src/db/shortcuts.ts

```diff
@@ -112,69 +151,85 @@ interface InsertSignatures {
 export const insert: InsertSignatures = function (
   table: Table,
   values: Insertable | Insertable[],
-  options?: ReturningOptionsForTable<Table, ColumnsOption<Table>, ExtrasOption<Table>>
+  options?: ReturningOptionsForTable<
+    Table,
+    ColumnsOption<Table>,
+    ExtrasOption<Table>
+  >
 ): SQLFragment<any> {
-
   let query;
   if (Array.isArray(values) && values.length === 0) {
     query = sql`INSERT INTO ${table} SELECT null WHERE false`;
     query.noop = true;
     query.noopResult = [];
-
   } else {
-    const
-      completedValues = Array.isArray(values) ? completeKeysWithDefaultValue(values, Default) : values,
-      colsSQL = cols(Array.isArray(completedValues) ? completedValues[0] : completedValues),
-      valuesSQL = Array.isArray(completedValues) ?
-        mapWithSeparator(completedValues as Insertable[], sql`, `, v => sql`(${vals(v)})`) :
-        sql`(${vals(completedValues)})`,
+    const postValues = applySerializeHook(table, values),
+      completedValues = Array.isArray(postValues)
+        ? completeKeysWithDefaultValue(postValues, Default)
+        : postValues,
+      colsSQL = cols(
+        Array.isArray(completedValues) ? completedValues[0] : completedValues
+      ),
+      valuesSQL = Array.isArray(completedValues)
+        ? mapWithSeparator(
+            completedValues as Insertable[],
```

Comment:
so I need to sort of ponder longer whether or not this is an issue (and how to test for it), but it seems like there is a potential type mismatch here

`values` is of type `Insertable | Insertable[]`, and it looks like the serialize hook also returns an `Insertable | Insertable[]` so I guess it is likely ok...but I guess I need to spend some time with the code and how it ensures that the values are appropriate for the table at hand. My guess is that the relationship between the values and table is handled at a higher level, and once it drops to this level it stops bothering to maintain that connection. regardless, assuming that zapatos ensures that relationship (it certainly does if you do `.select` or `.insert` via the generated types), I'm thinking about what happens if a serialize hooks converts into a type that is not appropriate for the table...I guess we just have to be careful at the hook level not to let that happen, in the same way that in the higher level `.select` etc APIs zapatos uses the generated code to try and avoid bad types getting through (since here it's all just `<any>`)